tests/models/sam2/test_modeling_sam2.py

```diff
@@ -0,0 +1,1422 @@
+        ann_frame_idx = 0  # the frame index we interact with
+        ann_obj_id = 1  # give a unique id to each object we interact with (it can be any integers)
+
+        inference_state = self.processor.process_new_points_or_box_for_video_frame(
+            inference_state=inference_state,
+            frame_idx=ann_frame_idx,
+            obj_ids=ann_obj_id,
+            input_points=[[[[210, 350]]]],
+            input_labels=[[[1]]],
+        )
+        outputs = self.model.infer_on_video_frame_with_new_inputs(
+            inference_state=inference_state,
+            frame_idx=ann_frame_idx,
+            obj_ids=ann_obj_id,
+            consolidate_at_video_res=False,  # Whether to save the masks at the video resolution (True) or at the model's resolution in the video session state (False)
+        )
+        low_res_masks, video_res_masks = outputs
+        self.assertEqual(low_res_masks.shape, (1, 1, 256, 256))
+        self.assertEqual(video_res_masks.shape, (1, 1, raw_video.shape[-3], raw_video.shape[-2]))
+        torch.testing.assert_close(
+            video_res_masks[0, 0, :3, :3],
+            torch.tensor(
+                [[-21.4113, -21.4113, -22.9685], [-23.3089, -23.3089, -24.2602], [-27.5700, -27.5700, -27.1607]]
+            ).to(torch_device),
+            atol=1e-4,
+            rtol=1e-4,
+        )
+
+        # test propagate in video frames
+        frames = []
+        for frame_idx, out_mask_logits in self.model.propagate_in_video(
+            inference_state=inference_state,
+            start_frame_idx=ann_frame_idx,
+            max_frame_num_to_track=2,
+        ):
+            frames.append(out_mask_logits)
+        frames = torch.stack(frames, dim=0)
+        self.assertEqual(frames.shape, (3, 1, 1, raw_video.shape[-3], raw_video.shape[-2]))
+        torch.testing.assert_close(
+            frames[:3, :, :, :2, :2],
+            torch.tensor(
+                [
+                    [[[[-21.4113, -21.4113], [-23.3089, -23.3089]]]],
+                    [[[[-20.0937, -20.0937], [-21.2233, -21.2233]]]],
+                    [[[[-19.9581, -19.9581], [-21.3028, -21.3028]]]],
+                ]
+            ).to(torch_device),
+            atol=1e-4,
+            rtol=1e-4,
+        )
+
+    def test_inference_mask_generation_video_multi_points(self):
+        raw_video = prepare_video()
+        inference_state = self.processor.init_video_session(video=raw_video, inference_device=torch_device)
+        ann_frame_idx = 0  # the frame index we interact with
+        ann_obj_id = 1  # give a unique id to each object we interact with (it can be any integers)
+
+        inference_state = self.processor.process_new_points_or_box_for_video_frame(
+            inference_state=inference_state,
+            frame_idx=ann_frame_idx,
+            obj_ids=ann_obj_id,
+            input_points=[[[[210, 350], [250, 220]]]],
+            input_labels=[[[1, 1]]],
+        )
```

Comment:
type annotations for input_points and input_labels are not correct, as well as return typehint